packages/troika-physics/src/facade/PhysicsManager.js

```diff
@@ -0,0 +1,394 @@
+        const volumePositions = geom.attributes.position.array
+        const volumeNormals = geom.attributes.normal.array
+        const association = geom.$physicsIndexAssociation
+
+        var numVerts = association.length
+        const flattenedDims = 6
+        for (let j = 0; j < numVerts; j++) {
+          var assocVertex = association[j]
+          const dj = j * flattenedDims
+          let x = nodes[dj + 0]
+          let y = nodes[dj + 1]
+          let z = nodes[dj + 2]
+          const nx = nodes[dj + 3]
+          const ny = nodes[dj + 4]
+          const nz = nodes[dj + 5]
+
+          sharedVec3.set(x, y, z)
+          facade.threeObject.worldToLocal(sharedVec3) // Translate world-space coords back to local
+
+          x = sharedVec3.x
+          y = sharedVec3.y
+          z = sharedVec3.z
+
+          for (var k = 0, kl = assocVertex.length; k < kl; k++) {
+            var indexVertex = assocVertex[k]
+
+            volumePositions[indexVertex] = x
+            volumeNormals[indexVertex] = nx
+            indexVertex++
+            volumePositions[indexVertex] = y
+            volumeNormals[indexVertex] = ny
+            indexVertex++
+            volumePositions[indexVertex] = z
+            volumeNormals[indexVertex] = nz
+          }
+        }
+
+        geom.attributes.position.needsUpdate = true
+        geom.attributes.normal.needsUpdate = true
+
+        if (!facade._matrixChanged) {
+          facade._matrixChanged = true
+        }
+        facade.afterUpdate()
+      }
+    }
+  }
+
+  _fireCollisionEvent (facadeId, otherFacadeId, contacts) {
+    // Fire onCollision event for each facade that registered for the event
+    const targetFacade = this._physicsObjectFacadesById[facadeId]
+    const collisionFacade = this._physicsObjectFacadesById[otherFacadeId]
+    if (!targetFacade || !collisionFacade) {
+      return
+    }
+    const newEvent = new CollisionEvent(
+      'collision', // eventType
+      targetFacade,
+      collisionFacade,
+      contacts
+    )
+    function callHandler (handler) {
+      handler.call(targetFacade, newEvent)
+    }
+    this.notifyWorld('dispatchEvent', {
+      targetFacade,
+      type: 'collision',
+      handler: callHandler
+    })
+  }
+
+  handleCollisionsUpdated (collisions) {
+    collisions.forEach(collision => {
+      const [facadeIdA, facadeIdB, contacts] = collision
+      this._fireCollisionEvent(facadeIdA, facadeIdB, contacts)
+      this._fireCollisionEvent(facadeIdB, facadeIdA, contacts)
+    })
+  }
+
+  handleBeforeRender () {
+    var deltaTime = this.clock.getDelta()
+
+    if (this._workerReady) {
+      if (this._needsPhysicsWorldUpdate) {
+        this._needsPhysicsWorldUpdate = false
+        this._updatePhysicsWorld()
+      }
+      if (this.simulationEnabled) {
+        this.sendToWorker('updatePhysicsWorld', [deltaTime])
+      }
+    }
+  }
+
+  // FIXME should this attempt to force the parent world into continuousRender mode?
```

Comment:
I think if you add a `this.notifyWorld('needsRender')` call in the worker response handler, that should give a similar result as continousRender.

Alternatively, you could consider having the physics manager set up its own timing loop rather than hooking in to onBeforeRender.

Reply:
Thanks for the suggestions, that definitely sounds cleaner. I'll investigate those options next.